frontend/src/screens/wallet/send/LnurlPay.tsx

```diff
@@ -0,0 +1,121 @@
+import React from "react";
+import { Button } from "src/components/ui/button";
+import { Input } from "src/components/ui/input";
+import { Label } from "src/components/ui/label";
+import { LoadingButton } from "src/components/ui/loading-button";
+import { useToast } from "src/components/ui/use-toast";
+
+import { LightningAddress, LnUrlPayResponse } from "@getalby/lightning-tools";
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { PayInvoiceResponse } from "src/types";
+import { request } from "src/utils/request";
+
+export default function LnurlPay() {
+  const { state } = useLocation();
+  const navigate = useNavigate();
+  const { toast } = useToast();
+
+  const lnurlDetails = state.args?.lnurlDetails as LnUrlPayResponse;
+  const [amount, setAmount] = React.useState("");
+  const [comment, setComment] = React.useState("");
+  const [isLoading, setLoading] = React.useState(false);
+
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      setLoading(true);
+      const lnAddress = new LightningAddress(lnurlDetails.identifier);
+      // this is set instead of calling fetch because
+      // requestInvoice uses lnurlpData.callback
+      lnAddress.lnurlpData = lnurlDetails;
+      const invoice = await lnAddress.requestInvoice({
+        satoshi: parseInt(amount),
+        comment,
+      });
+      const payInvoiceResponse = await request<PayInvoiceResponse>(
```

Comment:
This payment is duplicated with confirm payment. We could route to the ConfirmPayment page with the invoice instead - then the user can also confirm (make sure they didn't accidentally type the wrong amount)

Reply:
Hmmm, yeah we can do this to DRY it up even more!